slack_sdk/socket_mode/builtin/internals.py

```diff
@@ -204,8 +204,6 @@ def receive(specific_buffer_size: Optional[int] = None):
         size = specific_buffer_size if specific_buffer_size is not None else receive_buffer_size
         with sock_receive_lock:
             received_bytes = sock.recv(size)
-            if not received_bytes:
```

Comment:
This looks good to me, based on the [docs of this recv](https://docs.python.org/3/library/socket.html#socket.socket.recv) the connection should retry before throwing an exception, and the [behavior in _fetch_messages](https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/socket_mode/builtin/internals.py#L238) should properly handle the situation where the bytes are `None` or of length 0 🥇 

But let me know if I miss understood the intent of the change

Reply:
Thanks for the input @eddyg 💯  I was able to find the [source code of this behavior](https://github.com/python/cpython/blob/main/Lib/ssl.py#L1076) it does seem like enabling `suppress_ragged_eofs` could allow us to sort actual EOF from SSL_ERROR_EOF

Adding some sort of retry logic in [_fetch_messages](https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/socket_mode/builtin/internals.py#L238) when a SSL_ERROR_EOF is raised could be a good idea, what do you think @seratch?

Reply:
Note that `suppress_ragged_eofs` _is_ enabled by default in Python, and causes an improperly-terminated SSL session to just return "EOF".

By _disabling_ it in the SSLContext, `SSLEOFError`s would propagate out to this code. But as I mentioned, there may be consequences in doing so... there's likely a reason suppressing ragged EOFs defaults to `True` in Python.

It _may_ be interesting to disable it in a test app and see if `SSLEOFError`s _are_ seen communicating with the Slack server pipeline (my understanding is that it is actually quite common to terminate a session without properly tearing down the SSL session inside... that's probably why this parameter defaults to `True`!)

Anyway, this suggestion may be nothing more than a red herring, because as I mentioned, I'm not sure why the two-lines being proposed for removal were added in the first place! (Apologies for adding confusion!)

Reply:
@eddyg @WilliamBergamin Thanks a lot for you both's comments here! I think it's fine to have this revert but we may add/change the internals more in the future for the betterment of EOF and other pattern handling!